### PR TITLE
Approach

### PR DESCRIPTION
**Pull request by dmurvihill**
Fix latest Lauer comments

### CODE REVIEW

File: doc/src/approach.ltx [hunk 1]
```diff
@@ -244,16 +244,18 @@ collisions. Even a trivial change (a single letter added or removed, for
...
-changes to the output. Therefore, it is impossible to tell the degree of
-similarity between two tokens simply by comparing their hashes. This makes it
-impossible to identify similar lines just from their hashes, preventing 
-textit{Line Comparison} from being used to identify very similar hashes. Line
-Comparison remains in \textit{Checksims} both as a proof of concept and
-example of a simple algorithm, and to quickly identify extremely similar or
-identical submissions (an initial check that can be run prior to a slower
-but more accurate algorithm such as \textit{Smith-Waterman}).
+algorithm, unless preprocessors --- which cannot easily thwart sophisticated
+obfuscations such as changes to the abstract syntax tree --- are applied to
```
Comment (mheon) [inline doc/src/approach.ltx]: I don't like the "changes to the abstract syntax tree" example. How about "like arithmetic operand reordering"?

File: doc/src/approach.ltx [hunk 2]
```diff
@@ -244,16 +244,18 @@ collisions. Even a trivial change (a single letter added or removed, for
...
+combat them. Furthermore, almost all hashes enforce the property that any
+change in the input results in major changes to the output. Therefore, it is
+impossible to tell the degree of similarity between two tokens simply by
+comparing their hashes. Therefore, it is impossible to identify similar lines
+just from their hashes, preventing  textit{Line Comparison} from being used
```
Comment (mheon) [inline doc/src/approach.ltx]: Missing a backslash before textit

File: doc/src/approach.ltx [hunk 3]
```diff
@@ -269,13 +271,14 @@ using \textit{Line Comparison} with character tokenization.
...
-our test corpus. For more information about Smith-Waterman's performance, see
-Section~\ref{sec:results}. TODO add refs to specific sections discussing SW
+Section~\ref{sec:smithwatermanlit}. \textit{Smith-Waterman}'s first published
+use in academic dishonesty detection was by Robert Irving \cite{irving04}. When
+run against our test corpus, Smith-Waterman defeated multiple methods of
+obfuscation described in Section~\ref{sec:obfuscation}, including changed
+identifier names. We believe \textit{Smith-Waterman} caught most of the
+instances of academic dishonesty in our test corpus. For more information
+about Smith-Waterman's performance, see Section~\ref{sec:results}.
+TODO add refs to specific sections discussing SW
```
Comment (mheon) [inline doc/src/approach.ltx]: Recommend putting the TODO in a comment.

File: doc/src/methodology.ltx [hunk 4]
```diff
@@ -64,26 +64,25 @@ written to remove such information and generate usable test code.
...
-caveat that a professor manually review anonymized code to ensure that no
-obvious identifying information remained. Professor Lauer graciously
-offered to perform this final vetting.
+The first, and easiest to remove, were the filenames of the students' submission
+directories, which contained the usernames of the submitters. Simply renaming
+the directories was enough to remove this as a concern. Comments were the
+next concern, containing the vast majority of remaining personally
+identifiable information. Identifying information like names and usernames
+almost never occurred outside of comments. After consulting with several
+professors, it was decided that stripping comments and submission names was
```
Comment (mheon) [inline doc/src/methodology.ltx]: Lauer made it clear it wasn't from consulting with several professors, it was from consulting with the department head

File: doc/src/methodology.ltx [hunk 5]
```diff
@@ -158,59 +157,62 @@ In doing so, we would also obtain a figure for the number of false positives
...
-are especially helpful for testing the overall functionality of Checksims.
-They have been incorporated into the project as unit tests for individual
-algorithms, providing a verifiable set of results we can test for; they were
-not used in full-scale functional testing.
-
+These simulated corpora only exist for natural language, however, where our
+search is for similarity in source code. We are not confident that results
+obtained by testing Checksims against these corpora will correspond to our
+tool's performance against source code. However, we did use them in the
+project as unit tests for individual algorithms, providing a verifiable set
```
Comment (mheon) [inline doc/src/methodology.ltx]: Recommend replacing unit test with functional test

File: doc/src/approach.ltx (hunk 1 above)
Comment (dmurvihill) [inline doc/src/approach.ltx]: That's much better.

**Comment by dmurvihill** [comment]
All comments addressed.

**Pull request merged**